marlowe-dashboard-client/src/Capability/Marlowe.purs

```diff
@@ -460,18 +492,9 @@ instance monadMarloweHalogenM :: (ManageMarlowe m, ManageWebsocket m) => ManageM
   followContract walletDetails marloweParams = lift $ followContract walletDetails marloweParams
   createPendingFollowerApp = lift <<< createPendingFollowerApp
   followContractWithPendingFollowerApp walletDetails marloweParams followAppId = lift $ followContractWithPendingFollowerApp walletDetails marloweParams followAppId
-  createContract walletDetails roles contract = do
-    result <- lift $ createContract walletDetails roles contract
-    callMainFrameAction $ PlayAction $ Play.UpdateFromStorage
-    pure result
-  applyTransactionInput walletDetails marloweParams transactionInput = do
-    result <- lift $ applyTransactionInput walletDetails marloweParams transactionInput
-    callMainFrameAction $ PlayAction $ Play.UpdateFromStorage
-    pure result
-  redeem walletDetails marloweParams tokenName = do
-    result <- lift $ redeem walletDetails marloweParams tokenName
-    callMainFrameAction $ PlayAction $ Play.UpdateFromStorage
-    pure result
+  createContract walletDetails roles contract = lift $ createContract walletDetails roles contract
+  applyTransactionInput walletDetails marloweParams transactionInput = lift $ applyTransactionInput walletDetails marloweParams transactionInput
+  redeem walletDetails marloweParams tokenName = lift $ redeem walletDetails marloweParams tokenName
```

Comment:
We need to call `UpdateFromStorage` in these cases, but I'm now calling it explicitly at the relevant places in the action handlers instead. For two reasons: On second thoughts, it felt clearer (I actually forgot that I'd put this in here, and found myself wondering how the hell things were being updated). More importantly, following #3365, the old code didn't work as intended with `applyTransactionInput`, because it triggered the update _before_ the `mPendingTransaction` was set - so the `mPendingTransaction` wasn't getting cleared when it should have been.

(It took my a while to identify that bug. :weary:)